homeassistant/helpers/entity.py

```diff
@@ -518,8 +515,8 @@ async def async_internal_added_to_hass(self) -> None:
         if self.registry_entry is not None:
             assert self.hass is not None
             self.async_on_remove(
-                self.hass.bus.async_listen(
-                    EVENT_ENTITY_REGISTRY_UPDATED, self._async_registry_updated
+                async_track_entity_registry_updated_event(
+                    self.hass, self.entity_id, self._async_registry_updated
```

Comment:
We can clean up `_async_registry_updated` since we know that the entity matched, we can drop the `data["entity_id"] == self.entity_id` from the remove action handling. We can also drop all `create` actions (should not be possible I think, but better to drop them)

Reply:
done